File: iam/lib/iam.ex
defmodule IAM do
  def hello do
    %AWS.Client{
      access_key_id: System.get_env("ACCESS_KEY"),
      secret_access_key: System.get_env("ACCESS_SECRET"),
      region: "eu-central-1",
      endpoint: "amazonaws.com"
    }
    |> IO.inspect
    |> AWS.IAM.list_roles(%{})
    |> IO.inspect
  end
end

File: ruby/iam/lib/iam.rb
require 'aws-sdk-iam'

class IAM
  def hello
    creds = Aws.config[:credentials] = Aws::Credentials.new(
      ENV['ACCESS_KEY'], ENV['ACCESS_SECRET']
    )
    iam = Aws::IAM::Client.new
    puts iam.list_roles.map(&:arn)
    nil
  end
end

i = IAM.new
i.hello
